web_m2x_options/static/src/js/form.js

```diff
@@ -136,13 +136,14 @@ odoo.define('web_m2x_options.web_m2x_options', function (require) {
                 'ilike', this.limit + 1,
                 self.build_context()));
 
-            var create_rights;
             if (!(self.options && (self.is_option_set(self.options.create) || self.is_option_set(self.options.create_edit)))) {
-                create_rights = new Model(this.field.relation).call(
+                this.create_rights = this.create_rights || (function(){
+                return new Model(self.field.relation).call(
                     "check_access_rights", ["create", false]);
+                })();
             }
 
-            $.when(search_result, create_rights).then(function (data, can_create) {
+            $.when(search_result, this.create_rights).then(function (data, can_create) {
```

Comment:
I feel like it could happen that `this.create_rights` can be `undefined` in some circumstances (if the above `if` never executes). Would that be a problem?